pkg/providers/instance/azure_client.go

```diff
@@ -70,14 +73,23 @@ func CreateAzClient(cfg *auth.Config) (*AZClient, error) {
 }
 
 func NewAZClient(cfg *auth.Config, env *azure.Environment) (*AZClient, error) {
-	authorizer, err := auth.NewAuthorizer(cfg, env)
-	if err != nil {
-		return nil, err
+	var cred azcore.TokenCredential
+	var err error
+
+	if cfg.DeploymentMode == "managed" {
+		cred, err = azidentity.NewDefaultAzureCredential(nil)
```

Comment:
It's just a question: what's kind of credential will be used by function `azidentity.NewDefaultAzureCredential`? and how to mount this credential into gpu-provisioner pod?

Reply:
The managed mode will use the cluster identety that will be mounted to to container using a msi side-car

Reply:
Should we pass cluster identity for the function `azidentity.NewDefaultAzureCredential`, or nil is enough?